src/python/zquantum/core/measurement.py

```diff
@@ -347,31 +347,47 @@ def get_parities_from_measurements(
 
     # Count number of occurrences of bitstrings
     bitstring_frequencies = Counter(measurements)
+    bitstring_counts: np.ndarray = np.fromiter(
+        bitstring_frequencies.values(), dtype=int
+    )
 
     # Count parity occurrences
     values = []
     for _, term in enumerate(ising_operator.terms):
-        values.append([0, 0])
         marked_qubits = [op[0] for op in term]
-        for bitstring, count in bitstring_frequencies.items():
-            if check_parity(bitstring, marked_qubits):
-                values[-1][0] += count
-            else:
-                values[-1][1] += count
+        parity = check_parity_of_vector([*bitstring_frequencies.keys()], marked_qubits)
+
+        bitstrings_with_true_parity: int = (parity * bitstring_counts).sum()
```

Comment:
[suggestion] This variable is not really `bitstrings` –  it's an integer.
I would suggest renaming it to `true_parity_count` or sth like that.